src/Model/Behavior/ObfuscateBehavior.php

```diff
@@ -88,6 +89,10 @@ public function afterSave(Event $event, EntityInterface $entity, ArrayObject $op
      */
     public function beforeFind(Event $event, Query $query, ArrayObject $options, $primary)
     {
+        if (!isset($options['obfuscate'])) {
```

Comment:
This should be `if (empty($options['obfuscate']))`.  Also you can combine the `!$primary` check.